gprofiler/profilers/java.py

```diff
@@ -1017,18 +1033,33 @@ def stop(self) -> None:
     def _proc_exit_callback(self, tid: int, pid: int, exit_code: int) -> None:
         # Notice that we only check the exit code of the main thread here.
         # It's assumed that an error in any of the Java threads will be reflected in the exit code of the main thread.
-        if tid in self._profiled_pids:
+        if tid in self._want_to_profile_pids:
             self._pids_to_remove.add(tid)
+            java_version_output = self._pid_to_java_version.get(tid)
 
             signo = java_exit_code_to_signo(exit_code)
             if signo is None:
                 # not a signal, do not report
                 return
 
-            logger.warning("async-profiled Java process exited with signal", pid=tid, signal=signo)
+            if tid in self._profiled_pids:
+                logger.warning(
+                    "async-profiled Java process exited with signal",
+                    pid=tid,
+                    signal=signo,
+                    java_version_output=java_version_output,
+                )
 
-            if is_java_fatal_signal(signo):
-                self._disable_profiling(JavaSafemodeOptions.PROFILED_SIGNALED)
+                if is_java_fatal_signal(signo):
+                    self._disable_profiling(JavaSafemodeOptions.PROFILED_SIGNALED)
+            else:
+                # this is a process that we wanted to profile, but didn't profile due to safemode.
```

Comment:
could also be short lived and exit by the time you try to start AP

Reply:
```suggestion
                # this is a process that we wanted to profile, but didn't profile due to safemode / any other reason.
```